test/e2e/kubelet_stats.go

```diff
@@ -287,6 +356,102 @@ func formatResourceUsageStats(nodeName string, containerStats map[string]*contai
 	return fmt.Sprintf("Resource usage on node %q:\n%s", nodeName, buf.String())
 }
 
+type int64arr []int64
+
+func (a int64arr) Len() int           { return len(a) }
+func (a int64arr) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a int64arr) Less(i, j int) bool { return a[i] < a[j] }
+
+func computePercentiles(timeSeries map[time.Time]resourceUsagePerContainer, percentilesToCompute []int) map[int]resourceUsagePerContainer {
+	cpuData := make(map[string][]float64)
```

Comment:
I think that introducing
type A struct {
  x []float64
  y []float64
  z []float64
}
and having map[string]A would make this code simpler.